onmt/Loss.py

```diff
@@ -103,6 +104,106 @@ def compute_loss(self, batch, output, target, **kwargs):
 
         return loss, stats
 
+    # TODO should the weights be used to affect "num_correct"?
+    def stats(self, loss, scores, target):
+        """
+        Compute and return a Statistics object.
+
+        Args:
+            loss(Tensor): the loss computed by the loss criterion.
+            scores(Tensor): a sequence of predict output with scores.
+        """
+        pred = scores.max(1)[1]
+        non_padding = target.ne(self.padding_idx)
+        num_correct = pred.eq(target) \
+                          .masked_select(non_padding) \
+                          .sum()
+        return onmt.Statistics(loss[0], non_padding.sum(), num_correct)
+
+
+class NMTLossComputeDatumWeighted(LossComputeBase):
+    """
+    Standard NMT Loss Computation.
+    """
+    def __init__(self, generator, tgt_vocab):
+        super(NMTLossComputeDatumWeighted, self).__init__(generator, tgt_vocab)
+        self.copy_attn = False
+        weight = torch.ones(len(tgt_vocab))
+        weight[self.padding_idx] = 0
+        self.criterion = DatumWeightedNLLCriterion(weight, size_average=False)
+
+    def compute_loss(self, batch, output, target, **kwargs):
```

Comment:
For this concrete subclass, we don't want a vague `kwargs` here,  just list all args here it needs, not default `None` needed,  because if we use this `NMTLossComputeDatumWeighted`,  we already make sure these args are satisfied. 
You can refer to `onmt/CopyGenerator.py::CopyGeneratorLossCompute`.